### PR TITLE
Manpage: Replace `*` with `\*` in OPTIONS

### LINKED ISSUE #5108: Man page generation mis-parses pairs of asterisks as emphasis
This paragraph in `README.md` has two asterisks in it:

```
    --download-sections REGEX       Download only chapters whose title matches
                                    the given regular expression. Time ranges
                                    prefixed by a "*" can also be used in place
                                    of chapters to download the specified range.
                                    Needs ffmpeg. This option can be used
                                    multiple times to download multiple
                                    sections, e.g. --download-sections
                                    "*10:15-inf" --download-sections "intro"
```

After the man page is generated by sending the text through `devscripts/prepare_manpage.py` and then `pandoc` (invoked in the `Makefile`), those asterisks are gone and the text between them is emphasized; it looks like this:

scrot-12022-10-02-004418

### PR DESCRIPTION
**Pull request by leoheitmannruiz**
### Description of your *pull request* and other information

In the section "USAGE AND OPTIONS" in the README there's a few `*`. As this section is converted to Pandoc's definition_lists (no longer code blocks), the `*` are interpreted as emphasis.

This PR adds code to prepare_manpage.py, which replaces all occurrences of `*` with `\*`, so that they are properly escaped and displayed as intended in the manpage.

Fixes #5108

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [x] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

### CODE REVIEW

File: devscripts/prepare_manpage.py [hunk 1]
```diff
@@ -65,8 +65,12 @@ def move_sections(readme):
 
 def filter_options(readme):
     section = re.search(r'(?sm)^# USAGE AND OPTIONS\n.+?(?=^# )', readme).group(0)
+
+    # Replace * with \*. See https://github.com/yt-dlp/yt-dlp/issues/5108
+    section_new = section.replace('*', r'\*') if section else ''
```
Comment (pukkandan) [inline devscripts/prepare_manpage.py:70]:
```suggestion
    section = section.replace('*', R'\*')
```

File: devscripts/prepare_manpage.py [hunk 2]
```diff
@@ -65,8 +65,12 @@ def move_sections(readme):
...
+
     options = '# OPTIONS\n'
-    for line in section.split('\n')[1:]:
+    for line in section_new.split('\n')[1:]:
```
Comment (pukkandan) [inline devscripts/prepare_manpage.py:73]:
```suggestion
    for line in section.split('\n')[1:]:
```

**Review by pukkandan — APPROVED** [review]

**Review by bashonly — APPROVED** [review]

**Comment by bashonly** [comment]
`options` is generated from `section`

**Comment by bashonly** [comment]
Gotcha. whoops. go ahead and revert b787f9a41e8856173b7d5d42adc0810a86c556f1

File: devscripts/prepare_manpage.py [hunk 3]
```diff
@@ -65,8 +65,10 @@ def move_sections(readme):
 
 def filter_options(readme):
     section = re.search(r'(?sm)^# USAGE AND OPTIONS\n.+?(?=^# )', readme).group(0)
+    section_new = section.replace('*', r'\*')
```
Comment (bashonly) [inline devscripts/prepare_manpage.py:68]:
```suggestion
    section_new = section.replace('*', R'\*')
```
I think the uppercase in the original suggestion was to indicate a non-regex raw string

**Review by bashonly — APPROVED** [review]

**Comment by Grub4K** [comment]
Merged as df0e138fc02ae2764a44f2f59fc93c756c4d3ee2

**Pull request closed without merging**